src/utils/transformAwareTranslator.ts

```diff
@@ -12,41 +12,33 @@ export function getTranslator(component: Component): Translator {
   // The Translator works by first calculating the offset to root of the chart and then calculating
   // the offset from the component to the root. It is imperative that the _measurementElement
   // be added to the root of the hierarchy and nowhere else.
-  const root = component.root().rootElement().node() as Element;
+  const rootContent = component.root().content();
 
-  let translator: Translator = (<any> root)[_TRANSLATOR_KEY];
+  let translator: Translator = (<any> rootContent)[_TRANSLATOR_KEY];
   if (translator == null) {
-    const measurer = <SVGElement> <any>document.createElementNS(root.namespaceURI, "svg");
-
-    measurer.setAttribute("class", "measurer");
-    measurer.setAttribute("style", "opacity: 0; visibility: hidden; position: absolute; width: 1px; height: 1px;");
-
-    root.appendChild(measurer);
-
-    translator = new Translator(d3.select(measurer));
-    (<any> root)[_TRANSLATOR_KEY] = translator;
+    translator = new Translator(rootContent);
+    (<any> rootContent)[_TRANSLATOR_KEY] = translator;
   }
 
   return translator;
 }
 
-/**
- * Applies position as a style and attribute to the svg element
- * as the position of the element varies by the type of parent.
- * When nested within an SVG, the attribute position is respected.
- * When nested within an HTML, the style position is respected.
- */
-function move(node: SimpleSelection<any>, x: number, y: number) {
-  node.styles({ left: `${x}px`, top: `${y}px` });
-  node.attrs({ x: `${x}`, y: `${y}` });
-}
-
 export class Translator {
   private static SAMPLE_DISTANCE = 100;
-  private _measurementElement: SimpleSelection<void>;
-
-  constructor(measurementElement: SimpleSelection<void>) {
-    this._measurementElement = measurementElement;
+  private _measurer: d3.Selection<SVGRectElement, any, any, any>;
+
+  constructor(rootContent: SimpleSelection<void>) {
+    // Use a "rect" element inside the root content node to resolve https://github.com/palantir/plottable/issues/3355
```

Comment:
this comment is helpful but i would like a comment on the class explaining what the `Translator` does (i.e. append a `<rect>`) and why that's necessary for browser support.

Especially because constructing a new instance has DOM side-effects.